packages/core/src/utils/arrays.ts

```diff
@@ -1,11 +1,17 @@
 import { AnyLevelDeep } from "../types/arrays.js";
 
-export function flatten<T>(values: AnyLevelDeep<T>): T[] {
+type Flatten<T> = T extends unknown[] ? Flatten<T[number]> : T[];
```

Comment:
TIL there's a `FlatArray` type builtin. Unfortunately, I couldn't finagle it to work here.